src/DynamoCore/Graph/Workspaces/PackageDependencyInfo.cs

```diff
@@ -101,14 +101,139 @@ public override string ToString()
         }
     }
 
+    /// <summary>
+    /// Class containing info about a LocalDefinition
+    /// </summary>
+    internal class LocalDefinitionInfo: INodeLibraryDependencyInfo
+    {
+        /// <summary>
+        /// Name of the LocalDefinition
+        /// </summary>
+        public string Name { get; private set; }
+
+        /// <summary>
+        /// Path of the LocalDefinition
+        /// </summary>
+        public string Path { get; internal set; }
+
+        /// <summary>
+        /// Size of the LocalDefinition
+        /// </summary>
+        public string Size { get; internal set; }
+
+        /// <summary>
+        /// ReferenceType of the LocalDefinition
+        /// </summary>
+        public ReferenceType ReferenceType { get; internal set; }
+
+        private HashSet<Guid> nodes;
+
+        public HashSet<Guid> Nodes
+        {
+            get { return nodes; }
+        }
+
+        public Version Version { get; internal set; }
+
+        /// <summary>
+        /// Indicates whether this LocalDefinition is loaded in the current session
+        /// </summary>
+        [Obsolete("This property is obsolete", false)]
+        public bool IsLoaded { get; set; }
+
+        public PackageDependencyState State { get; internal set; }
+
+        /// <summary>
+        /// Create a LocalDefinition info object from the package Name and Path
+        /// </summary>
+        /// <param name="name"></param>
+        /// <param name="path"></param>
+        internal LocalDefinitionInfo(string name, string path)
+        {
+            Name = name;
+            Path = path;
+            nodes = new HashSet<Guid>();
+        }
+
+        /// <summary>
+        /// Create a LocalDefinition info object from the package Name and ReferenceType
+        /// </summary>
+        /// <param name="name"></param>
+        /// <param name="referenceType"></param>
+        internal LocalDefinitionInfo(string name, ReferenceType referenceType)
+        {
+            Name = name;
+            ReferenceType = referenceType;
+            nodes = new HashSet<Guid>();
+        }
+
+        /// <summary>
+        /// Create a LocalDefinition info object from the package name
+        /// </summary>
+        /// <param name="name"></param>
+        internal LocalDefinitionInfo(string name)
+        {
+            Name = name;
+            nodes = new HashSet<Guid>();
+        }
+
+        /// <summary>
+        /// Checks whether two LocalDefinitionInfo's are equal
+        /// They are equal if their Name and Versions are equal
+        /// </summary>
+        /// <param name="obj"></param>
+        /// <returns></returns>
+        public override bool Equals(object obj)
+        {
+            if (obj == null)
+            {
+                return false;
+            }
+            if (!(obj is LocalDefinitionInfo))
+            {
+                return false;
+            }
+
+            var other = obj as LocalDefinitionInfo;
```

Comment:
```
var other = obj as LocalDefinitionInfo;
if(other == null) return false;

if (other.Name == this.Name && other.Path == this.Path) 
{ ... }
```

Reply:
How about also checking for the reference type to be equal?

Reply:
I will be creating another PR for external references and will make this change in that PR.